tests/src/org.eclipse.jetty/jetty-server/12.0.1/src/test/java/jetty/JettyTests.java

```diff
@@ -0,0 +1,120 @@
+ */
+package jetty;
+
+import org.eclipse.jetty.ee10.servlet.ServletContextHandler;
+import org.eclipse.jetty.ee10.webapp.WebAppContext;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.TypeUtil;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledInNativeImage;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.time.Duration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JettyTests {
+
+    private static final int PORT = 8080;
+
+    private static boolean DEBUG = false;
+
+    @BeforeAll
+    static void beforeAll() {
+        System.setProperty("org.slf4j.simpleLogger.defaultLogLevel", DEBUG ? "debug" : "warn");
+    }
+
+    @Test
+    void typeUtilWorks() {
+        assertThat(TypeUtil.fromName("java.lang.String")).isEqualTo(String.class);
+    }
+
+    @Test
+    void http() throws Exception {
+        Server server = new Server(PORT);
+        server.setHandler(new Handler.Abstract() {
+            @Override
+            public boolean handle(Request request, Response response, Callback callback) {
+                response.setStatus(200);
+                response.getHeaders().add("Content-Type", "text/plain");
+                response.write(true, ByteBuffer.wrap("Hello world".getBytes(StandardCharsets.UTF_8)), callback);
+                return true;
+            }
+        });
+        server.start();
+        try {
+            HttpResponse<String> response = doHttpRequest();
+            assertThat(response.statusCode()).isEqualTo(200);
+            assertThat(response.body()).isEqualTo("Hello world");
+        } finally {
+            server.stop();
+        }
+    }
+
+    @Test
+    void servlet() throws Exception {
+        Server server = new Server(PORT);
+        ServletContextHandler handler = new ServletContextHandler();
+        handler.setContextPath("/");
+        handler.addServlet(HelloWorldServlet.class, "/*");
+        server.setHandler(handler);
+        server.start();
+        try {
+            HttpResponse<String> response = doHttpRequest();
+            assertThat(response.statusCode()).isEqualTo(200);
+            assertThat(response.body()).isEqualTo("Hello world");
+        } finally {
+            server.stop();
+        }
+    }
+
+    @Test
+    // This test doesn't work in a native image and fails with:
+    // java.lang.IllegalArgumentException: not an allowed scheme: resource:/org/eclipse/jetty/ee10/webapp/webdefault-ee10.xml
+    @DisabledInNativeImage
```

Comment:
This test doesn't work in a native image. It runs on the JVM, but fails in a native image. The reason is the `resource` handler in GraalVM. On the JVM, this is starting with `file:` or `jar:`, which is on an allow-list in Jetty. `resource:` is not on there and therefore something inside Jetty breaks. This can't be solved with reflection-metadata and needs to be tackled in Jetty itself.

Reply:
Related to https://github.com/eclipse/jetty.project/issues/9116

Reply:
This can be workaround with this small snippet:

```
ResourceFactory.registerResourceFactory("resource", new URLResourceFactory());
```

Reply:
Not sure if we want that snippet in our tests. I've added a 2nd commit which adds this snippet, but feel free to drop it.